test/build.spec.ts

```diff
@@ -226,14 +226,57 @@ describe('Resolved project building', () => {
 			const resolveListeners = {
 				error: [reject],
 			};
-			const newTask = resolveTask(task, 'x86', 'intel-nuc', resolveListeners);
+			const newTask = resolveTask(task, 'amd64', 'intel-nuc', resolveListeners);
 			resolve(runBuildTask(newTask, docker, secretMap, buildVars));
-		}).then((image: LocalImage) => {
-			expect(image)
-				.to.have.property('successful')
-				.that.equals(true);
-			return checkExists(image.name!);
-		});
+		})
+			.then((image: LocalImage) => {
+				expect(image)
+					.to.have.property('successful')
+					.that.equals(true);
+				return checkExists(image.name!);
+			})
+			.then((inspect: any) => {
+				expect(inspect)
+					.to.have.property('Architecture')
+					.that.equals('amd64');
+			});
+	});
+
+	it('should correctly build a resolved project for a different platform', function() {
+		if (
+			docker.version().then((version: any) => {
+				return version.experimental;
```

Comment:
Is this missing some `await` clause? Because these lines would return immediately with a Promise object, with would the considered truthy and then the `if` test would always succeed. Also, the PR at the moment removed `experimental` detection, so this test should probably never be skipped. (?)

Reply:
so I just prepend an `await`?

Reply:
```
		if (
			await docker.version().then((version: any) => {
				return version.experimental;
			})
		) {
			this.skip();
		}

```

Reply:
I guess we can also just drop this check, no?

Reply:
since we're not going to do that anyway...

Reply:
we can't drop it

Reply:
the engine used by circleci is pretty old...
``
Server Engine Details:
  Version:          17.09.0-ce
  API version:      1.32 (minimum version 1.12)
  Go version:       go1.8.3
  Git commit:       afdb6d4
  Built:            2017-09-26T22:40:56.000000000+00:00
  OS/Arch:          linux/amd64
  Experimental:     false
``